core/services/ocr2/plugins/ocr2vrf/coordinator/proxy.go

```diff
@@ -0,0 +1,82 @@
+package coordinator
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/pkg/errors"
+
+	evmclient "github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/core/gethwrappers/generated"
+	"github.com/smartcontractkit/chainlink/core/gethwrappers/ocr2vrf/generated/vrf_beacon"
+	"github.com/smartcontractkit/chainlink/core/gethwrappers/ocr2vrf/generated/vrf_coordinator"
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+var _ VRFBeaconCoordinator = &VRFProxy{}
+
+//go:generate mockery --name VRFBeaconCoordinator --output ./mocks/ --case=underscore
+
+// VRFProxy routes requests to VRFBeacon and VRFCoordinator go wrappers and implements VRFBeaconCoordinator interface
+type VRFProxy struct {
+	lggr        logger.Logger
+	beacon      vrf_beacon.VRFBeacon
+	coordinator vrf_coordinator.VRFCoordinator
```

Comment:
It also might be better that these refer to the interfaces (I think you can search for `VRFBeaconInterface` and `VRFCoordinatorInterface`) generated by the gowrappers so that this code can be tested in unit tests (although it's a simple enough component, but doesn't hurt to add tests).

Reply:
makes sense. changed to interface

Reply:
added unit tests also!